modules/reboot/src/main/java/org/mule/runtime/module/reboot/internal/SystemMuleVersionBootstrapConfigurer.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.reboot.internal;
+
+import static java.lang.String.format;
+import static java.lang.System.setProperty;
+
+import org.mule.runtime.module.reboot.api.MuleContainerBootstrapUtils;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.Date;
+import java.util.Properties;
+
+/**
+ * A {@link BootstrapConfigurer} which takes care of setting the mule.version system properties based on the given POM file.
+ */
+public class SystemMuleVersionBootstrapConfigurer implements BootstrapConfigurer {
```

Comment:
Taken from [MuleContainerBootstrap](https://github.com/mulesoft/mule/pull/12618/files#diff-8e506ff6291a77a23727e462ed729a5ca0bf6b9c77f669390026275495a49b67L126), but refactored the error handling. It now uses a try-with-resources too.